pkg/v1/providers/ytt/03_customizations/02_avi/add-avi-labels.yaml

```diff
@@ -0,0 +1,20 @@
+#@ load("@ytt:data", "data")
+#@ load("@ytt:json", "json")
+#@ load("@ytt:overlay", "overlay")
+#@ load("@ytt:template", "template")
+
+
+#! Label workload cluster if AVI_LABELS is specified
+
+#@ if data.values.PROVIDER_TYPE == "vsphere" and data.values.TKG_CLUSTER_ROLE == "workload" and data.values.AVI_LABELS:
```

Comment:
either cluster_option model file's possible values for TKG_CLUSTER_ROLE needs to be updated or the check here should be to allow some test cases to trigger this overlay.
When I tried that, I am noticing that the overlay does not work as expected when the config file has the AVI_LABELS set as this (with no quotes)

```
...
AVI_LABELS: {foo: 'bar'}
```

error seen:
```
- template.replace: expected value to be map, but was string
    in <toplevel>
      03_customizations/02_avi/add-avi-labels.yaml:18 |     _: #@ template.replace(data.values.AVI_LABELS)
```

PTAL to see if the format of the value or overlay needs to be updated.

Reply:
@vuil  I've flipped the logic into `if data.values.TKG_CLUSTER_ROLE != "management"`

Also, for `AVI_LABELS` to work, the value should be surrounded by the double quotation, which the current testcase gen cannot produce. For example,

```
AVI_LABELS: "{foo: 'bar'}"
```

Reply:
Can we merge this one first as it's blocking other tasks? We can have a separate thread on how to fix the testcase gen.